src/platform/src/main/java/org/geoserver/platform/resource/Resources.java

```diff
@@ -711,7 +711,7 @@ public static URL toURL(final Resource res) {
                     "resource",
                     null,
                     -1,
-                    String.format(res.getType() == Type.DIRECTORY ? "/%s/" : "/%s", res.path()),
+                    String.format(res.getType() == Type.DIRECTORY ? "%s/" : "%s", res.path()),
```

Comment:
Is this going to change anything for existing data directories?

Reply:
This changes what is being passed around internally at runtime (to make clear difference between (File, Path, and Location).

The internal resource URL was introducing a ``/`` prefix here presenting an odd situation where the generated path would get double slash ``//`` as part of the URL. The processing of Path was covering for this by ignoring the empty ``""`` directory name between the two slashes ``//``.

By removing this double slash the code is easier to follow.